torchaudio/prototype/pipelines/source_separation_pipeline.py

```diff
@@ -0,0 +1,93 @@
+from dataclasses import dataclass
+from functools import partial
+from typing import Callable
+
+import torch
+import torchaudio
+
+from torchaudio.prototype.models import conv_tasnet_base
+
+
+class _Separator(torch.nn.Module):
+    def __init__(self, model: torch.nn.Module):
+        super().__init__()
+        self.model = model
+
+    def forward(self, feature: torch.Tensor) -> torch.Tensor:
+        """Separates the input mixture speech into different sources.
+
+        Args:
+            input (torch.Tensor): input tensor with dimensions `[batch, time]`.
```

Comment:
Is it possible to generalize shape for future inputs that could include things like num_channels